pkg/controller/build/reconciler.go

```diff
@@ -191,6 +228,285 @@ func (b *buildReconciler) deleteMachineOSConfig(ctx context.Context, mosc *mcfgv
 	return nil
 }
 
+func checkAndInstallPipeline(ctx context.Context, kubeclient clientset.Interface, pipelineoperatorclient pipelineoperatorclientset.Interface, olmclient olmclientset.Interface, tektonclient tektonclientset.Interface, postBuildTasks []string) error {
+	tektonNamespace := "openshift-pipelines"
+	operatorsNamespace := "openshift-operators"
+	tektonConfigName := "config"
+	subscriptionName := "openshift-pipelines-operator"
+	tektonPipelineName := "build-and-push-pipeline"
+	tektonClusterTaskName := "buildah"
+	var namespaceDNE, tektonconfigDNE, tektonPipelineDNE, buildahTaskDNE bool = false, false, false, false
+
+	// Ensure "openshift-pipelines" Namespace exists
+	_, err := kubeclient.CoreV1().Namespaces().Get(ctx, tektonNamespace, metav1.GetOptions{})
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			namespaceDNE = true
+		} else {
+			return fmt.Errorf("openshift-pipelines namespace get error %v", err)
+		}
+	}
+
+	// Ensure TektonConfig exists
+	_, err = pipelineoperatorclient.OperatorV1alpha1().TektonConfigs().Get(ctx, tektonConfigName, metav1.GetOptions{})
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			tektonconfigDNE = true
+		} else {
+			return fmt.Errorf("tektonConfig resource get error %v", err)
+		}
+	}
+	if namespaceDNE || tektonconfigDNE {
+		// Define the Subscription resource
+		subscription := &olmv1alpha1.Subscription{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "operators.coreos.com/v1alpha1",
+				Kind:       "Subscription",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      subscriptionName,
+				Namespace: "openshift-operators",
+			},
+			Spec: &olmv1alpha1.SubscriptionSpec{
+				Channel:                "latest",
+				Package:                "openshift-pipelines-operator-rh",
+				CatalogSource:          "redhat-operators",
+				CatalogSourceNamespace: "openshift-marketplace",
+			},
+		}
+		_, err = olmclient.OperatorsV1alpha1().Subscriptions(operatorsNamespace).Create(ctx, subscription, metav1.CreateOptions{})
+		if err != nil {
+			if k8serrors.IsAlreadyExists(err) {
+				klog.V(2).Infof("%v already exists", subscriptionName)
+			} else {
+				return fmt.Errorf("subscription resource create error %v", err)
+			}
+		}
+	}
+	interval := 1 * time.Minute
+	timeout := 20 * time.Minute
+	err = waitForTektonConfigReady(ctx, pipelineoperatorclient, tektonNamespace, tektonConfigName, interval, timeout)
+	if err != nil {
+		return err
+	}
+
+	// Ensure pre-built buildah task exists
+	existingBuildah, err := tektonclient.TektonV1beta1().Tasks(ctrlcommon.OpenshiftPipelinesNamespace).Get(ctx, tektonClusterTaskName, metav1.GetOptions{})
```

Comment:
thought (non-blocking): For each of these tasks and pipeline definitions, we may want to do something like:

1. Check if it exists. If not, create it.
2. If it exists, ensure that it has not changed from our definition. If it has, delete it and recreate it.